core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpHeaders.java

```diff
@@ -0,0 +1,199 @@
+package com.linecorp.armeria.internal.common;
+
+import static com.linecorp.armeria.internal.common.ArmeriaHttpUtil.COOKIE_SEPARATOR;
+import static com.linecorp.armeria.internal.common.ArmeriaHttpUtil.COOKIE_SPLITTER;
+import static com.linecorp.armeria.internal.common.ArmeriaHttpUtil.HTTP_TO_HTTP2_HEADER_DISALLOWED_LIST;
+import static com.linecorp.armeria.internal.common.ArmeriaHttpUtil.convertHeaderValue;
+import static com.linecorp.armeria.internal.common.ArmeriaHttpUtil.toHttp2HeadersFilterTE;
+import static com.linecorp.armeria.internal.common.ArmeriaHttpUtil.toLowercaseMap;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.RequestHeadersBuilder;
+import com.linecorp.armeria.internal.common.ArmeriaHttpUtil.CaseInsensitiveMap;
+
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.util.AsciiString;
+
+public final class ArmeriaHttpHeaders extends HttpHeaders {
```

Comment:
We can put this method in the package `com.linecorp.armeria.server` and remove the public modifier.